### PR TITLE
Refactor code around sms fragments in redis

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Refactor the code around the sms counter to make
- the redis counter always equal the number of fragments sent
- make the code much easier to read / use

My devcontainer is being flakey 😞  but it looks like this might have fixed
https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/912

Should verify this in staging though.

### CODE REVIEW

File: app/sms_fragment_utils.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    if current_app.config["REDIS_ENABLED"]:
+        fetch_daily_sms_fragment_count(service_id)  # to make sure it's set in redis
+        cache_key = sms_daily_count_cache_key(service_id)
+        redis_store.incrby(cache_key, increment_by)
+
+
+def delete_daily_sms_fragment_count(service_id: UUID):
+    if current_app.config["REDIS_ENABLED"]:
+        cache_key = sms_daily_count_cache_key(service_id)
+        redis_store.delete(cache_key)
```
Comment (andrewleith) [inline app/sms_fragment_utils.py:33]: Nice! So clean! 🧹

File: app/service/rest.py [hunk 2]
```diff
@@ -297,7 +297,7 @@ def update_service(service_id):
         if not fetched_service.restricted:
             _warn_service_users_about_message_limit_changed(service_id, current_data)
     if sms_limit_changed:
-        redis_store.delete(sms_daily_count_cache_key(service_id))
+        delete_daily_sms_fragment_count(service_id)
```
Comment (andrewleith) [inline app/service/rest.py:300]: Why does this get deleted when the limit changes?  Doesn't this number represent the number of fragments sent today?

File: app/service/rest.py [hunk 3]
```diff
@@ -297,7 +297,7 @@ def update_service(service_id):
...
         redis_store.delete(near_sms_daily_limit_cache_key(service_id))
```
Comment (andrewleith) [inline app/service/rest.py:301]: `near_sms_daily_limit_cache_key` and `over_sms_daily_limit_cache_key` don't appear to be used anywhere in the API - Are they for future enhancements?

File: app/service/rest.py (hunk 2 above)
Comment (sastels) [reply app/service/rest.py:300]: yeah.... I had copied the code around the general daily limit, but that cache key is called "daily_limit_cache_key" and it looks like Antoine thought it was the cache key for the daily limit (silly Antoine) and so he added code here to delete it https://github.com/cds-snc/notification-api/pull/1258

But that's just a mistake caused by our bad naming

File: app/service/rest.py (hunk 3 above)
Comment (sastels) [reply app/service/rest.py:301]: We use them to check if we've already sent the service owners the warning / alert emails
https://github.com/cds-snc/notification-api/blob/c8393182334506d1ec0d0648319bed6569021ea4/app/notifications/validators.py#L180-L181
https://github.com/cds-snc/notification-api/blob/c8393182334506d1ec0d0648319bed6569021ea4/app/notifications/validators.py#L202-L203

**Review by andrewleith — APPROVED** [review]
LGTM!

**Pull request merged**